src/views/ItemListView.vue

```diff
@@ -287,27 +289,30 @@ onBeforeMount(() => {
 });
 
 onMounted(() => {
-    doRequest(`${apiBaseUrl}${route.fullPath}`, () => {
-        defaultProfile.value = ui.profiles[profiles.value.find(p => p.default)!.uri];
-        
-        // if specify mediatype, or profile is not default or alt, redirect to API
-        if ((route.query && route.query._profile) &&
-            (route.query._mediatype || ![defaultProfile.value.token, ALT_PROFILES_TOKEN].includes(route.query._profile as string))) {
-                window.location.replace(`${apiBaseUrl}${route.path}?_profile=${route.query._profile}${route.query._mediatype ? `&_mediatype=${route.query._mediatype}` : ""}`);
-        }
+    loading.value = true;
+    ensureProfiles().then(() => {
+        doRequest(`${apiBaseUrl}${route.fullPath}`, () => {
```

Comment:
Would it make sense to call the ensureProfiles in App.vue? That way, the code that performs the checking is done in one place instead of having to do it within one of the view components.

Reply:
Resolving - side discussion needed to work out common pattern with accessing central store